### DISCUSSION TITLE
A few feature suggestions...

### DISCUSSION THREAD

**Discussion by FlaMtnBkr** [OrcaSlicer/OrcaSlicer#7778] [Ideas]
Was wondering about once a model is sliced, where you can go line by line and it shows the line number and the height, if the printed weight could be added? I think the bottom can also be rolled up so then there could be the weight of the sliced object between the 2 lines? Also wondering how hard it would be to add the center of mass?

Another thing I've noticed on longer prints, is that filament will start to build up on the nozzle which is always an invitation for something to stick and get pulled around. Was wondering if a nozzle wipe at the back of the printer (not on the model) after a set amount of time, like every 5 min or whatever you choose. Or it might be easier if it's after every 15 Layers etc so it doesn't stop in the middle of a layer?

Just some thoughts that I thought might be helpful to people? Most of my posts never get replies so I must not have very interesting ideas or problems....

**Comment by Anthony-Bec** [comment]
+1

**Comment by DARKFiB3R** [comment]
Just out of interest, why would you want to have such granular information about the weight? Thanks

**Reply by hexsmith** [reply]
The slicer knows the filament used at any given moment.  The volume can be calculated easily, and a user-specified density value could give you this information rather easily.

**Reply by FlaMtnBkr** [reply]
Mainly to find the center of gravity more easily. 

I mainly print functional parts and not many figurines and toys though I do some for my daughter. It's easy to find the COG in CAD when parts have a uniform density. But when the walls, infill density, and material change it makes it harder to figure out. For most things it doesn't matter at all, but for some models and prototyping, it could be a convenient feature. I also don't think it would be hard to implement so why not?!

Also, you can change wall count and infill and slice a part and it might be nice to know the difference in weight of the section that has the biggest change in weight from changes in settings. The top and bottom doesn't usually change much, at least with flat sides, so the biggest change in weight will happen between certain layers. Again, it should be easy to calculate, so why not.
